src/utils/arrangerUtils.js

```diff
@@ -147,24 +149,35 @@ export const findValueInField = (source, path, defaultValue = null) => {
  * @returns {object[]}
  */
 export const reduceAndMerge = (source, path) => {
-  const pathSegments = path.split('.');
-  return deepReduceAndMerge(source, source, pathSegments);
+    const pathSegments = path.split('.');
+    return deepReduceAndMerge(source, source, pathSegments);
 };
 
 const deepReduceAndMerge = (source, currentLevel, segments) => {
-  const propName = segments.slice(0, 1);
-  const prop = currentLevel[propName];
-
-  if (!Array.isArray(prop)) {
-    return [currentLevel];
-  }
-
-  return prop.reduce((result, item) => {
-    return result.concat(
-      deepReduceAndMerge(source, item, segments.slice(1)).map(t => ({
-        ...currentLevel,
-        [propName]: t,
-      })),
-    );
-  }, []);
-};
+    const currentPathName = segments.slice(0, 1);
```

Comment:
can you put some documentation here? Kind of code your lost when your back after a break.

Also, it did not exactly what I would expect it to do (based on name and other behavior like reduce which normally returns a value, not an array)

e.g. in test
```
const result = reduceAndMerge({
            a: 1,
            b: [{c: [{d: 2}, {d: 4}, {d: null}]}],
            e: 4
        }, 'b.c.d');
        expect(result).to.be.eql([
            {a: 1, b: {c: {d: 2}}, e: 4},
            {a: 1, b: {c: {d: 4}}, e: 4},
            {a: 1, b: {c: {d: null}}, e: 4},
        ]);
```
Instead I'm expecting something like 
```
const result = reduceAndMerge({
         ...
        expect(result).to.be.eql(
            {a: 1, b: {c: { d: [2, 4, null ] }, e: 4}
        );
```

Reply:
I had the same thought. I changed the name...Not sure it's better but a least it is less misleading.